[Doc][KubeRay] Improve RayJob batch inference example (#42785)
ray-project/ray: doc/source/cluster/kubernetes/examples/rayjob-batch-inference-example.md

# RayJob Batch Inference Example

This example demonstrates how to use the RayJob custom resource to run a batch inference job for an image classification workload on a Ray cluster.

> See the document [Image Classification Batch Inference with HuggingFace Vision Transformer](https://docs.ray.io/en/latest/data/examples/huggingface_vit_batch_prediction.html) for a full explanation of the code.

Reviewer: ```suggestion
See [Image Classification Batch Inference with HuggingFace Vision Transformer](https://docs.ray.io/en/latest/data/examples/huggingface_vit_batch_prediction.html) for a full explanation of the code.
```

## Step 1: Install the KubeRay Operator

Follow [this document](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.

> It should be scheduled on the CPU pod.

Author: "CPU pod" is wrong.

## Step 2: Submit the RayJob

Create the RayJob custom resource with [ray-job.batch-inference.yaml](https://github.com/ray-project/kuberay/blob/v1.0.0/ray-operator/config/samples/ray-job.batch-inference.yaml).

Download the file with `curl`:

```bash
curl -LO https://raw.githubusercontent.com/ray-project/kuberay/v1.0.0/ray-operator/config/samples/ray-job.batch-inference.yaml
```

> Note that the `RayJob` spec contains a spec for the `RayCluster` that is to be created for the job. For this tutorial, we use a single-node cluster with 4 GPUs. For production use cases, we recommend using a multi-node cluster where the head node doesn't have GPUs, so that Ray can automatically schedule GPU workloads on worker nodes and they won't interfere with critical Ray processes on the head node.

Reviewer: ```suggestion
Note that the `RayJob` spec contains a spec for the `RayCluster` that is to be created for the job. This tutorial uses a single-node cluster with 4 GPUs. For production use cases, use a multi-node cluster where the head node doesn't have GPUs, so that Ray can automatically schedule GPU workloads on worker nodes which won't interfere with critical Ray processes on the head node.
```

Reviewer: Can you change "is to be created for the job" to active voice? I don't know who is doing the creating so I don't have a quick suggestion for you.

Author: I directly remove "that is to be created for the job.". It is not important.

[...]
Status:
  Dashboard URL:          rayjob-sample-raycluster-j6t8n-head-svc.default.svc.cluster.local:8265
  End Time:               ...

>   Job Deployment Status:  Complete

Author: In the master branch, if JobStatus is SUCCEEDED, the JobDeploymentStatus will eventually transition to Complete.

  End Time:               ...
  Job Deployment Status:  Complete
  Job Id:                 rayjob-sample-ft8lh
  Job Status:             SUCCEEDED
  Message:                Job finished successfully.
  Observed Generation:    2

>   Ray Cluster Name:       rayjob-sample-raycluster-j6t8n

Author: This is useless.
